pandas/core/internals.py

```diff
@@ -632,9 +633,13 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
 
             return self.make_block(Categorical(self.values, dtype=dtype))
 
+        # convert dtypes if needed
+        try:
+            dtype = pandas_dtype(dtype)
+        except TypeError:
+            pass
```

Comment:
I'm missing the rationale for not raising here

Reply:
hmm, good point. I think I can take this out, previously this was ``np.dtype`` which would raise on lots of things (like pandas dtypes). ``pandas_dtype`` only raises on completely invalid things.